internal/supervisor/installation.go

```diff
@@ -77,31 +77,46 @@ type installationProvisioner interface {
-	provisioner                        installationProvisioner
-	aws                                aws.AWS
-	instanceID                         string
+	store             installationStore
+	provisioner       installationProvisioner
+	aws               aws.AWS
+	instanceID        string
+	keepDatabaseData  bool
+	keepFilestoreData bool
+	scheduling        InstallationSupervisorSchedulingOptions
+	resourceUtil      *utils.ResourceUtil
+	logger            log.FieldLogger
+}
+
+// InstallationSupervisorSchedulingOptions are the various options that control
+// how installation scheduling occurs.
+type InstallationSupervisorSchedulingOptions struct {
+	balanceInstallations               bool
 	clusterResourceThreshold           int
 	clusterResourceThresholdScaleValue int
-	keepDatabaseData                   bool
-	keepFilestoreData                  bool
-	resourceUtil                       *utils.ResourceUtil
-	logger                             log.FieldLogger
 }
 
 // NewInstallationSupervisor creates a new InstallationSupervisor.
-func NewInstallationSupervisor(store installationStore, installationProvisioner installationProvisioner, aws aws.AWS, instanceID string, threshold, thresholdScaleValue int, keepDatabaseData, keepFilestoreData bool, resourceUtil *utils.ResourceUtil, logger log.FieldLogger) *InstallationSupervisor {
+func NewInstallationSupervisor(store installationStore, installationProvisioner installationProvisioner, aws aws.AWS, instanceID string, keepDatabaseData, keepFilestoreData bool, scheduling InstallationSupervisorSchedulingOptions, resourceUtil *utils.ResourceUtil, logger log.FieldLogger) *InstallationSupervisor {
 	return &InstallationSupervisor{
-		store:                              store,
-		provisioner:                        installationProvisioner,
-		aws:                                aws,
-		instanceID:                         instanceID,
-		clusterResourceThreshold:           threshold,
-		clusterResourceThresholdScaleValue: thresholdScaleValue,
-		keepDatabaseData:                   keepDatabaseData,
-		keepFilestoreData:                  keepFilestoreData,
-		resourceUtil:                       resourceUtil,
-		logger:                             logger,
+		store:             store,
+		provisioner:       installationProvisioner,
+		aws:               aws,
+		instanceID:        instanceID,
+		keepDatabaseData:  keepDatabaseData,
+		keepFilestoreData: keepFilestoreData,
+		scheduling:        scheduling,
+		resourceUtil:      resourceUtil,
+		logger:            logger,
+	}
+}
+
+// NewInstallationSupervisorSchedulingOptions creates a new InstallationSupervisorSchedulingOptions.
+func NewInstallationSupervisorSchedulingOptions(balanceInstallations bool, clusterResourceThreshold, clusterResourceThresholdScaleValue int) InstallationSupervisorSchedulingOptions {
+	return InstallationSupervisorSchedulingOptions{
+		balanceInstallations:               balanceInstallations,
```

Comment:
Is it possible that we may wish to have other scheduling strategies? 

0/5 but I wonder if this field should be something like `strategy` as an enum-like type (maybe an iota or just typed const strings) with set values to represent the two current strategies (`naive` (instead of a `false` value), and `balanced` (instead of `true)) with an eye towards experimentation with other strategies (like a random scheduler, or round robin, or something more intelligent that we haven't thought of yet)

Reply:
Yeah, we discussed this idea a bit above in Joram's comment. Definitely something we may want, but will just go with the simpler config for now and will revisit in the future.